tests/foreman/api/test_contentcredentials.py

```diff
@@ -260,14 +260,13 @@ def test_positive_block_delete_key_in_use(module_org, target_sat, capfd):
     assert repo.gpg_key.id == gpg_key.id
     assert product.gpg_key.id == repo.gpg_key.id
 
-    # Attempt to delete gpg in use
-    with pytest.raises(HTTPError) as err:
-        gpg_key.delete()
-
-    # Check for 500 error and display message
-    err_message = capfd.readouterr()[1]
-    assert '500' in str(err.value)
-    assert 'Cannot delete' in str(err_message)
+    # Attempt to delete gpg in use, capturing api response without raising exception
+    response = gpg_key.delete_raw()
+    assert response.status_code == 500
+    assert 'errors' in str(response.json())
+    assert 'Cannot delete record because of dependent root_repositories' in str(
+        response.json()['errors']
+    )
```

Comment:
We can combine these two checks together by setting a default value that will fail the second assertion, if `errors` isn't present.
```suggestion
    assert 'Cannot delete record because of dependent root_repositories' in str(
        response.json().get('errors')
    )
```